src/onegov/activity/iso20022.py

```diff
@@ -38,11 +38,11 @@ def __repr__(self):
 
     @cached_property
     def references(self):
+        if self.reference:
+            return {self.reference}
         return set(self.extract_references())
 
     def extract_references(self):
-        if self.reference:
-            yield self.reference
 
         # currently only one schema supports text extraction, the others are
         # stored in the designated reference field
```

Comment:
It was creating unique references based on the `booking_text`, which was always the same. 
This was the reason for transactions incorrectly being labelled as duplicates.

Reply:
Does this really work with other schemas than feriennet-v1? ESR/QR-References are obviously stored in other fields...

Reply:
Failing test supports this suspicion...

Reply:
Maybe the logic "reference field present -> must be a reference schema" is flawed. Maybe it should be more something like references contains all possible references (ESR and things from the note).

Reply:
I don't have a test file for `feriennet-v1` so it' hard to say.